liquibase-extension-testing/src/main/java/liquibase/extension/testing/testsystem/TestSystem.java

```diff
@@ -38,12 +41,49 @@ public abstract class TestSystem implements TestRule, Plugin {
     ));
 
     static {
-        //cache configured test systems for faster lookup
         configuredTestSystems = Scope.getCurrentScope().getSingleton(LiquibaseConfiguration.class).getCurrentConfiguredValue(ConfigurationValueConverter.STRING, null, "liquibase.sdk.testSystem.test").getValue();
-        if (configuredTestSystems != null) {
-            for (String definition : StringUtil.splitAndTrim(configuredTestSystems, ","))
-                testSystems.add(TestSystem.Definition.parse(definition));
+        skippedTestSystems = Scope.getCurrentScope().getSingleton(LiquibaseConfiguration.class).getCurrentConfiguredValue(ConfigurationValueConverter.STRING, null, "liquibase.sdk.testSystem.skip").getValue();
+
+        for (String definition : determineEnabledTestSystems(configuredTestSystems, skippedTestSystems)) {
```

Comment:
It's probably better to isolate the lookup of all the test systems to only when they have "all" as a value? Instead of always looking them up and then only using it in one case?

That also gets rid of the need for this method vs. the one below that doesn't expose having to construct all testSystems before calling?

Reply:
Good idea.